### PR TITLE
feat: allow slack message template as function

### PR DESCRIPTION
**Pull request by AndrewLeedham**
WIP Behaviour for discussed changes in https://github.com/juliuscc/semantic-release-slack-bot/pull/39

### CODE REVIEW

File: lib/fail.js [hunk 1]
```diff
@@ -27,13 +27,22 @@ module.exports = async (pluginConfig, context) => {
...
-  if (pluginConfig.onFailTemplate) {
-    slackMessage = template(pluginConfig.onFailTemplate, {
+  if (failTemplate) {
+    const variables = {
       package_name,
       repo_path: repoPath,
       repo_url: repoURL
-    })
+    }
+    if (typeof failTemplate === 'object') {
```
Comment (BeyondEvil) [inline lib/fail.js:38]: You can probably replace this with a case statement. Will probably be cleaner.

Example [here](https://github.com/juliuscc/semantic-release-slack-bot/blob/master/lib/template.js)

**Comment by BeyondEvil** [comment]
Would you mind adding an example of what the `releaserc.js` file would look like?

File: lib/success.js [hunk 2]
```diff
@@ -42,15 +42,26 @@ module.exports = async (pluginConfig, context) => {
...
+  if (successTemplate) {
+    const variables = {
       package_name,
       npm_package_version: nextRelease.version,
       repo_path: repo.path,
       repo_url: repo.URL,
       release_notes: releaseNotes
-    })
+    }
+    if (typeof successTemplate === 'object') {
```
Comment (BeyondEvil) [inline lib/success.js:55]: Same.

**Comment by AndrewLeedham** [comment]
> Would you mind adding an example of what the `releaserc.js` file would look like?

I am going to add tests and documentation when I get some free time. This is still a WIP.

File: lib/fail.js (hunk 1 above)
Comment (AndrewLeedham) [reply lib/fail.js]: Personally I prefer the `if else` syntax because breaks make switches confusing. But I don't have that strong of a preference so I updated it to use blocked switches. (blocked because the indented syntax is hard to follow)

**Comment by juliuscc** [comment]
I've completely forgot about this PR. Is this covered by the other PR's that we merged before?

**Comment by AndrewLeedham** [comment]
Yeah, I believe the other PR covers this.
